scipy/stats/tests/test_stats.py

```diff
@@ -5952,6 +5952,21 @@ def test_masked_3d_array(self):
         assert_equal(gstd_actual.mask, mask)
 
 
+@pytest.mark.parametrize('alternative', ['two-sided', 'greater', 'less'])
+def test_binom_test_deprecation(alternative):
+    deprecation_msg = ("'binom_test' is deprecated in favour of"
+                       " 'binomtest' from version 1.7.0 and will"
+                       " be removed in Scipy 1.12.0.")
+    num = 10
+    X = np.random.randint(10, 100, (num,))
+    N = X + np.random.randint(0, 100, (num,))
+    P = np.random.uniform(0, 1, (num,))
```

Comment:
Since this test is just to verify that we raise the proper warning, we can use simple small arrays defined by hand instead of using random (side note, this would have required using `np.random.Generator` instead.)

Reply:
And could you please be more specific about `np.random.Generator`?

Reply:
Right so you need to do:

```suggestion
    rng = np.random.default_rng(156114182869662948677852568516310985853)
    X = rng.integers(10, 100, (num,))
    N = X + rng.integers(0, 100, (num,))
    P = rng.uniform(0, 1, (num,))
```

Reply:
Is there any point of doing that? I mean, if new and old values does not always coincide one has more chances to spot it w/o fixing the seed.

Reply:
Yes, we want to be able to reproduce locally. Here it's just a smoke test since we are going to remove the function. If there is more concerns, we could add a loop, still using the generator. This way we would test the behaviour multiple times on various inputs.